wlm-ui/src/MainPage/ChannelList.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { channelListActions, selectChannelList } from "../store/slices/channel/channel";
+
+export default function ChannelListTable() {
+    const channelListState = useSelector(selectChannelList);
+    useEffect(() => {
+        localStorage.setItem('channel.channelList', JSON.stringify(channelListState.channels));
+    }, [channelListState]);
+
+    const dispatch = useDispatch();
+
+    const onClickUse = (channel: number) => {
+        return () => {
+            dispatch(channelListActions.toggleUse({ channel: channel }));
+        };
+    };
+
+    return (
+        <table>
+            <thead><tr>
+                <th>Use</th>
+                <th>Channel</th>
+                <th>Name</th>
+            </tr></thead>
+            <tbody>{channelListState.channels.map((chinfo) => {
+                const channel = chinfo.channel.channel;
+                return (
+                    <tr key={channel}>
+                        <td><button onClick={onClickUse(channel)}>
+                            {chinfo.inUse ? 'O' : 'X'}
+                        </button></td>
+                        <td>{channel}</td>
+                        <td>{chinfo.channel.name}</td>
+                    </tr>
+                );
+            })}</tbody>
+        </table>
```

Comment:
I think too many line breaking is not good, because it makes too many indentations.
So my suggestion is:
- Try to write a tag e.g., `<button ...>...</button>`, `<input .../>`, in a single line if it is possible.
- If a tag has only a single tag inside, try to write it also in a single line, e.g., `<td><button ...>...</button></td>`.
- If a line break is required because of line length, try not to break a tag itself e.g., `<button ...>`, but its content: `<button ...>(\n)...(\n)</button>`.

Of course feel free to suggest any other idea!

Reply:
I find it cool😃 Let's try to follow your suggestion.